servant-server/src/Servant/Server/Internal.hs

```diff
@@ -400,8 +403,9 @@ instance ( AllCTUnrender list a, HasServer sublayout
         -- http://www.w3.org/2001/tag/2002/0129-mime
         let contentTypeH = fromMaybe "application/octet-stream"
                          $ lookup hContentType $ requestHeaders request
-        mrqbody <- handleCTypeH (Proxy :: Proxy list) (cs contentTypeH)
-               <$> lazyRequestBody request
+        lbody <- lazyRequestBody request
+        mrqbody <- traverse (liftIO . unsafeInterleaveIO . runExceptT) $
```

Comment:
This is going to need some thought. I'm not sure it's safe. We do the `bodyCheck` possibly multiple times per request, and probably having the side-effect happen multiple times is not what people expect.